pages/profil.tsx

```diff
@@ -1,59 +1,104 @@
 import { withTransaction } from '@elastic/apm-rum-react'
 import { GetServerSideProps } from 'next'
-import React from 'react'
+import React, { ChangeEvent } from 'react'
 
-import { Conseiller, UserStructure } from 'interfaces/conseiller'
+import { Switch } from 'components/ui/Switch'
+import { UserStructure } from 'interfaces/conseiller'
 import { ConseillerService } from 'services/conseiller.service'
 import styles from 'styles/components/Layouts.module.css'
 import useMatomo from 'utils/analytics/useMatomo'
+import useSession from 'utils/auth/useSession'
 import { withMandatorySessionOrRedirect } from 'utils/auth/withMandatorySessionOrRedirect'
-import withDependance from 'utils/injectionDependances/withDependance'
+import { useConseiller } from 'utils/conseiller/conseillerContext'
+import { useDependance } from 'utils/injectionDependances'
 
 interface ProfilProps {
-  conseiller: Conseiller
   structureConseiller: string
   pageTitle: string
 }
 
-function Profil({ conseiller, structureConseiller }: ProfilProps) {
+function Profil({ structureConseiller }: ProfilProps) {
   const labelAgence =
     structureConseiller === UserStructure.MILO ? 'Mission locale' : 'agence'
 
+  const conseillerService =
+    useDependance<ConseillerService>('conseillerService')
+
+  const [conseiller, setConseiller] = useConseiller()
+  const { data: session } = useSession<true>({ required: true })
+
+  function toggleNotificationsSonores(e: ChangeEvent<HTMLInputElement>) {
+    if (conseiller && session) {
+      const conseillerMisAJour = {
+        ...conseiller,
+        notificationsSonores: Boolean(e.target.checked),
+      }
+      setConseiller(conseillerMisAJour)
+      conseillerService.modifierNotificationsSonores(
+        conseiller.id,
+        conseillerMisAJour.notificationsSonores,
+        session.accessToken
+      )
+    }
+  }
+
   useMatomo('Profil')
 
-  return (
+  return conseiller ? (
```

Comment:
```suggestion
  return conseiller && (
```
et retirer le ternaire, si pas de conseiller, on affiche aucune informations non?
sinon
return `return conseiller && (...)
!conseiller && <></>
`

Reply:
Vu avec @arthurlbrjc : on a fait une solution emprunte de sa proposition et de la sienne, tu me diras si ça te va :)